azure/services/aso/aso.go

```diff
@@ -188,3 +232,35 @@ func deepCopyOrNil(obj genruntime.MetaObject) genruntime.MetaObject {
 	}
 	return obj.DeepCopyObject().(genruntime.MetaObject)
 }
+
+// IsManaged returns whether the ASO resource referred to by spec was created by
+// CAPZ and therefore whether CAPZ should manage its lifecycle.
+func IsManaged(ctx context.Context, ctrlClient client.Client, spec azure.ASOResourceSpecGetter, clusterName string) (bool, error) {
+	ctx, _, done := tele.StartSpanWithLogger(ctx, "aso.Service.IsManaged")
+	defer done()
+
+	resource := spec.ResourceRef()
+	err := ctrlClient.Get(ctx, client.ObjectKeyFromObject(resource), resource)
+	if err != nil {
+		return false, errors.Wrap(err, "error getting resource")
+	}
+
+	return ownedByCluster(resource.GetLabels(), clusterName), nil
+}
+
+func ownedByCluster(labels map[string]string, clusterName string) bool {
+	return labels[infrav1.OwnedByClusterLabelKey] == clusterName
+}
+
+func mergeMaps[K comparable, V any](base map[K]V, overrides map[K]V) map[K]V {
```

Comment:
Is there no existing util out there in go libraries? If not, should probably go in a util package since it's generic / not directly related to ASO

Reply:
It was honestly easier to write this than to try to find something that already exists that works the same way. I think this is small enough that it's not worth looking too deep for. `golang.org/x/exp/maps` is the only place off the top of my head I would think already has something like this but it doesn't, and I can't think of a similar util that exists in a module we already import. I'll move this to `util/maps`at least.